drivers/gpu/drm/rp1/rp1-dpi/rp1_dpi.c

```diff
@@ -353,6 +354,20 @@ static int rp1dpi_platform_probe(struct platform_device *pdev)
 	if (ret)
 		goto done_err;
 
+	dpi->rgb_order_override = -1;
```

Comment:
My only comment is that this could potentially be defined as RP1DSPI_ORDER_UNCHANGED, and check for that by name instead of `< 0` at line 423.

Reply:
It's copied from vc4_dpi.c (though in that instance the values were from a register field).

BTW I also toyed with encoding the lookup order more directly in the variable (values like 0x020100) to avoid the switch statement.

Reply:
vc4_dpi.c isn't always the best example, and there is no requirement to copy how it is implemented in the code, only the DT side.

Reply:
Added `RP1DSI_ORDER_UNCHANGED` as an explicit value, and re-worked the lookup to use bit-shifts (which makes both object and source code slightly smaller. That last `& 3` is superstitious since the values are internal).